packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -765,17 +767,16 @@ class TextPainter {
       double newWidth;
       switch (textWidthBasis) {
         case TextWidthBasis.longestLine:
-          // The parent widget expects the paragraph to be exactly
-          // `TextPainter.width` wide, if that value satisfies the constraints
-          // it gave to the TextPainter. So when `textWidthBasis` is longestLine,
-          // the paragraph's width needs to be as close to the width of its
-          // longest line as possible.
           newWidth = _applyFloatingPointHack(_paragraph!.longestLine);
           break;
         case TextWidthBasis.parent:
           newWidth = maxIntrinsicWidth;
           break;
       }
+      // TODO(LongCatIsLooong): ui.Paragraph.width always reports the input
+      // width. The double layout keeps the two width in sync so TextPainter
+      // does not have to work with a non-zero paint offset. SkParagraph
+      // invalidates the layout entirely in this case so this is expensive.
```

Comment:
Should we create an issue for this?  Or maybe it's not possible or easy to reproduce a userland bug caused by this.

Reply:
This is one important use case that has to be covered in the new paragraph API. I'll describe that in the design doc instead.